src/definition/App.ts

```diff
@@ -197,6 +197,19 @@ export abstract class App implements IApp {
         return;
     }
 
+    /**
+     * Method which is called before a setting which belongs to this App is going to be updated
+     * by an external system and not this App itself. The setting passed is the newly updated one.
+     *
+     * @param setting the setting which is going to be updated
+     * @param configurationModify the accessor to modifiy the system
+     * @param reader the reader accessor
+     * @param http an accessor to the outside world
+     */
+    public async onPreSettingUpdate(setting: ISetting, configurationModify: IConfigurationModify, read: IRead, http: IHttp): Promise<ISetting> {
```

Comment:
I would think the setting's "old value" could be passed as a parameter here, maybe as a `SettingUpdateContext` object as a first parameter. The Apps-Engine already has this value in its scope, so providing it as a parameter of the hook will prevent apps from having to reach the DB to grab it